### ISSUE TITLE
IDE crashes while compiling an Inine recursive function with ToScopedNativeWChar!();

### ISSUE THREAD

**Issue by gopakumar-gg** [beefytech/Beef#1523]
IDE crashes while compiling the below code

```csharp
using System;

namespace Test
{
	class Program
	{
		public static void Main()
		{
			Recurse(1);
		}

		[Inline]
		public static char16* Recurse(int i)
		{
			var i;
			if (++i < 1)
				return Recurse(i);
			else
				return "abc".ToScopedNativeWChar!();
		}
	}
}
```

Crash occurs only with the combination  of Inline, Recursion, and call to ToScopedNativeWChar!();

1.   If you don't make it Inline, it won't crash.
2.   If you remove the recursion, it won't crash.
3.   I changed the return type to int and replaced  "abc".ToScopedNativeWChar!(); with a user-defined simple mixin call.  It did not crash.

**Comment by gopakumar-gg** [comment]
Originally, I wanted something similar to Win32 representation of unicode string literals with prefix L as in  ( wchar_t * ptr = L"abc").   So, I tried define an inline function L (string str).   It compiled successfully.   Then I wanted to make sure it is really Inline as Beef documentation promises.  I added some dummy code to introduce a recursion, expecting a compilation error saying it cannot be made inline because of recursion.   Instead of compilation error, it crashed.
          Then I set aside the Unicode attempts, and just checked if it would compile if I make a minimal most recursive function Inline (with Inline attribute).  To my surprise, it compiled successfully.   Doesn't Beef guarantee Inline?

**Comment by bfiete** [comment]
Fixed at 501491d87547b082d25052916f7369b9251e6170

Keep in mind that inlining a method does not mean the scope of things like `ToScopedNativeWChar` will extend to the invoking method. You need to use a mixin for that kind of thing.

And, no, recursive inlining does not cause an error - the original invocation will be inlined but the recursive calls just won't be.

**Comment by gopakumar-gg** [comment]
Thanks for the clarifications, Brian.
